session/pingpong/invoice_tracker.go

```diff
@@ -232,7 +232,10 @@ func (it *InvoiceTracker) listenForExchangeMessages() error {
 
 func (it *InvoiceTracker) generateAgreementID() {
 	agreementID := make([]byte, 32)
-	crand.Read(agreementID)
+	_, err := crand.Read(agreementID)
+	if err != nil {
+		panic(err)
```

Comment:
nothing useful to log before panic?

Reply:
nope, except panic error itself. I don't expect this happening anyway, just a measure of precaution.